Quick look into fighter stances. Does it matter or not?

In [1]:
import pandas as pd

In [2]:
data = pd.read_csv('../../data/ufc-master.csv')

In [12]:
different_stance = data['B_Stance'] != data['R_Stance']
#Looking fights where stances are different
data_filtered = data[different_stance][['B_Stance', 'R_Stance', 'Winner']]
print(len(data_filtered))

1673


We have 1673 fights where fighters used different stances. Now we have to check how many times fighter with an orthodox stance won and how many times fighter with an southpaw stance to determine if there is any difference.

In [16]:
#Because we have only 2 options and we know the number of fights where fighters had different stances we only have to find out
#number of wins for one stance and we can get the other one by subtraction
orthodox_wins = 0
for i in range(len(data_filtered)):
    if (data_filtered['B_Stance'].iloc[i] == 'Orthodox') & (data_filtered['Winner'].iloc[i] == 'Blue'):
        orthodox_wins += 1
    elif (data_filtered['R_Stance'].iloc[i] == 'Orthodox') & (data_filtered['Winner'].iloc[i] == 'Red'):
        orthodox_wins += 1
southpaw_wins = len(data_filtered) - orthodox_wins

In [23]:
print("When both fighters used different stances fighter with an orthodox stance (right handed) won: " +
     str(orthodox_wins) + "/" + str(len(data_filtered)) + " fights (" +
      str(round((orthodox_wins / len(data_filtered)) * 100, 2)) + "%).")
print("When both fighters used different stances fighter with an southpaw stance (left handed) won: " +
     str(southpaw_wins) + "/" + str(len(data_filtered)) + " fights (" +
      str(round((southpaw_wins / len(data_filtered)) * 100, 2)) + "%).")

When both fighters used different stances fighter with an orthodox stance (right handed) won: 767/1673 fights (45.85%).
When both fighters used different stances fighter with an southpaw stance (left handed) won: 906/1673 fights (54.15%).


This is an interesting result. As we can see when both fighters use different stances more often than not it is better to prefer fighter who uses a southpaw stance instead of a fighter who uses orthodox stance. The sample size was 1673, which tells us that this is certainly not random either and there is certainly some kind of correlation.
One of the reasons why this might be the case is that most of the population is right handed and because of this most of the time fighters are fighting against other right handed fighters. This means that left handed fighters are probably used to fighting agaisnt right hinded fighters, but right handed fighters do not get to fight against left handed fighters that often so the situation is less familiar for them.